moveit_ros/planning/rdf_loader/src/rdf_loader.cpp

```diff
@@ -115,13 +126,107 @@ rdf_loader::RDFLoader::RDFLoader(TiXmlDocument *urdf_doc, TiXmlDocument *srdf_do
     srdf_.reset(new srdf::Model());
     if (!srdf_->initXml(*urdf_, srdf_doc))
     {
-      ROS_ERROR("Unable to parse SRDF");
+      ROS_ERROR_NAMED("rdf_loader", "Unable to parse SRDF");
       srdf_.reset();
     }
   }
   else
   {
-    ROS_ERROR("Unable to parse URDF");
+    ROS_ERROR_NAMED("rdf_loader", "Unable to parse URDF");
     urdf_.reset();
   }
 }
+
+bool rdf_loader::RDFLoader::isXacroFile(const std::string& path)
+{
+  // TODO: implement case-insensitive search
+  return path.find(".xacro") != std::string::npos;
+}
+
+bool rdf_loader::RDFLoader::loadFileToString(std::string& buffer, const std::string& path)
+{
+  if (path.empty())
+  {
+    ROS_ERROR_NAMED("rdf_loader", "Path is empty");
+    return false;
+  }
+
+  std::ifstream stream(path.c_str());
+  if (!stream.good())
+  {
+    ROS_ERROR_NAMED("rdf_loader", "Unable to load path");
+    return false;
+  }
+
+  // Load the file to a string using an efficient memory allocation technique
+  stream.seekg(0, std::ios::end);
+  buffer.reserve(stream.tellg());
+  stream.seekg(0, std::ios::beg);
+  buffer.assign((std::istreambuf_iterator<char>(stream)), std::istreambuf_iterator<char>());
+  stream.close();
+
+  return true;
+}
+
+bool rdf_loader::RDFLoader::loadXacroFileToString(std::string& buffer, const std::string& path, const std::vector<std::string>& xacro_args)
+{
+  if (path.empty())
+    return false;
+
+  std::string cmd = "rosrun xacro xacro --inorder ";
```

Comment:
"Because --inorder processing is much more powerful, in future version beyond Jade, the new processing style will become default and you should check the compatibility of your xacro files."

Looks like in Kinetic they are still just giving a warning, but I think its safe to assume people will have fixed it in kinetic. I already made this assumption in the setup assistant: https://github.com/ros-planning/moveit/commit/a597058eb8fcb6d7dee410f0ee38783d5c5945c5